### DISCUSSION TITLE
Defining a study cohort using a medication based codelist

### DISCUSSION THREAD

**Discussion by SRW612** [opensafely/documentation#381] [Q&A] [answered]
I am trying to create a codelist of anitbiotics for acute conditions that can be used with the function _with_these_medications_ in the study definition. 

I have created a codelist using the pseudoBNF codebuilder on opencodelists.org, however, the python function to import codelists (_codelist_from_csv_) does not recognise pseudoBNF and so SNOMED dm+d codes must be used. 

I have tried downloading the dm+d version of my pseudoBNF codelist then using that to create a new dm+d codelist via CSV upload, but many of the codes are not recognised by either opencodelists.org or by the SNOMED CT browser. 

Manually looking up each code in the online browser does work to distinguish geniune codes, but I have 2000+ codes which is not feasible. 

Is there any simpler way to create a dm+d codelist in opencodelists.org that will actually work in my study definition program?

**Comment by inglesp** [comment, marked as answer]
Hi @SRW612, there's a gap in our documentation here, apologies.

If you go to https://www.opencodelists.org/codelist/user/[your-username]/add/ then you'll see a form that allows you to upload a dm+d codelist.  Let me know if you have any problems.

**Reply by SRW612** [reply]
Thanks @inglesp, that has worked. The only modification was that the CSV document needed column titles, where the usual CSV upload facility specifies no column titles. In any case, all seems to be working now.
